app/services/interface_merge.rb

```diff
@@ -0,0 +1,23 @@
+class InterfaceMerge
+  def run(interfaces, compute_attrs)
+    vm_interfaces = compute_attrs.vm_interfaces
+
+    # merge with existing
+    interfaces.each do |nic|
+      if !nic.virtual?
```

Comment:
maybe this would be more readable
```ruby
interfaces.select(&:physical?).each do |nic|
...
end
```

which requires adding
```ruby
def physical?
  !virtual?
end
```
to interface, this might be handy elsewhere